spec/models/tip_spec.rb

```diff
@@ -2,51 +2,158 @@
 
 describe Tip do
 
-  let(:user) { FactoryGirl.create(:user) }
-  before { @tip = user.tips.build(title: "Tip", body: "some things", description: "Lorem ipsum") }
+  let(:user) { FactoryGirl.build_stubbed(:user) }
+  let(:tip) { FactoryGirl.build_stubbed(:tip) }
 
-  subject { @tip }
+  subject { tip }
 
   it { should respond_to(:title) }
   it { should respond_to(:description) }
   it { should respond_to(:body) }
   it { should respond_to(:user_id) }
+  it { should respond_to(:slug) }
   it { should respond_to(:user) }
   
-  it { should be_valid }
+  describe 'when all validations are satisfied' do
+    it { should be_valid }
+  end
 
+  describe "associations" do
+    it { should belong_to(:user).counter_cache(true) }
+    it { should have_many(:bookmarks).dependent(:destroy) }
+  end
 
-  describe "when user_id is not present" do
-  	before { @tip.user_id = nil }
-  	it { should_not be_valid }
+  describe "before save" do
+    it "should pygmentize_and_cache_body" do
+      expect(tip).to receive(:pygmentize_and_cache_body)
+      tip.save! 
+    end
   end
 
   describe "with blank title" do
```

Comment:
Move all the validation tests up so they are 'grouped' with the other validations. Also, use the shoulda matchers that allow you to test the validation conditions without having to do the individual setups for each test. Check how I implemented these in the User model.